src/providers/Bee.tsx

```diff
@@ -209,6 +222,22 @@ export function Provider({ children }: Props): ReactElement {
     refresh()
   }, [beeDebugApi]) // eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    if (nodeAddresses?.ethereum) {
+      // debounced calls
+      const xdai = Rpc.eth_getBalance(nodeAddresses.ethereum)
+      const bzz = Rpc.eth_getBalanceERC20(nodeAddresses.ethereum)
+
+      if (xdai?.then) {
+        xdai.then(balance => setXdai(balance))
+      }
+
+      if (bzz?.then) {
+        bzz.then(balance => setBzz(balance))
+      }
+    }
+  }, [nodeAddresses])
+
```

Comment:
@vojtechsimetka It seemed a good idea to put the RPC balance checks in a `useEffect` hook depending on `nodeAddresses`. However that triggered many duplicate requests at the same time, so I had to introduce debouncing. Lmk what you think of this approach.

Reply:
Yeah this is good. But I think even putting `[nodeAddresses?.ethereum]` in the hook could solve it (so at the beginning it is null and then always the same address. But I am also ok with what you implemented.